js/main.js

```diff
@@ -27,6 +41,54 @@ function hideBackOnHome() {
 
 hideBackOnHome();
 
+function toggleFavorite(e) {
+  if (e.target.tagName === 'I' && e.target.className === 'fas fa-heart align-items-center favorite-icon inactive') {
+    e.target.className = 'fas fa-heart align-items-center favorite-icon active';
+    for (var i = 0; i < data.allDeals.length; i++) {
+      var newObj = {};
+      if (data.allDeals[i].dealID === e.target.closest('.newContainer').querySelector('[data-dealid]').getAttribute('data-dealid')) {
```

Comment:
You want to store the second value in this comparison in a variable. It makes readability a little difficult